### ISSUE TITLE
Outdated azure-storage version -> incompatible with newer azure-blob-storage

### ISSUE THREAD

**Issue by sklingel** [microsoft/dpu-utils#60]
We encountered some issues with incompatible versions of azure-blob-storage required by dpu_utils, e.g. with packages like mlflow.

Overwritting the azure-blob-storage version required by dpu_utils leads to the following error:
```
ImportError: cannot import name 'BlobClient' from 'azure.storage.blob' (/home/klingest/miniconda3/envs/ds-mlflow/lib/python3.8/site-packages/azure/storage/blob/__init__.py)
```

This could be fixed locally for us by replacing: https://github.com/microsoft/dpu-utils/blob/master/python/dpu_utils/utils/richpath.py#L22
with 
`from azure.storage.blob.blockblobservice import BlockBlobService`

Is this a fix that should perhaps go into the master?
Since it looks like there is not much more dependencies in the package of the "old" azure-storage package: https://github.com/microsoft/dpu-utils/search?q=azure as well as the deprecation of azure-storage, from my understanding

**Comment by mallamanis** [comment]
Hi @sklingel ,

It is odd that you have problem, `BlockBlobService` is imported at `__init__.py` of `azure.storage.blob`. It might have to do with miniconda? I either case, if your fix works, I'd be happy to change the code.

Upgrading to the latest version ([`azure-storage-blob`](https://pypi.org/project/azure-storage-blob/) needs some work. I had started this some time ago (here #62 ) but haven't got to finishing this.  Any help, would be appreciated.

**Comment by mallamanis** [comment]
Once #61 is merged, I'll push a new release in PyPI

**Comment by sklingel** [comment]
Ok, I think we have a misunderstanding. I manually installed a newer version of azure-blob-storage, since the version required by dpu_utils is too old to work with newer packages together in one (conda) environment. 
And it seems that the newer versions of azure-blob-storage need this different import I pointed out.

**Comment by mallamanis** [comment]
I see. Then we need to get #62 done at some point soon...

**Comment by mallamanis** [comment]
This should be now fixed at [0.3.1](https://pypi.org/project/dpu-utils/#history).

Let us know if you continue having problems.
